packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -103,84 +95,14 @@ export function SynapseChat({
       ? AgentAccessLevel.WRITE_YOUR_PRIVATE_DATA
       : AgentAccessLevel.PUBLICLY_ACCESSIBLE,
   )
-  // When both the current message and current response are available, add a new ChatInteraction to the array
-  const [localInteractions, setLocalInteractions] = useState<ChatInteraction[]>(
-    [],
-  )
-  const interactions = externalInteractions ?? localInteractions
-  const setInteractions = setExternalInteractions ?? setLocalInteractions
 
-  const [pendingInteraction, setPendingInteraction] =
-    useState<ChatInteraction>()
-  const [currentlyProcessingJobId, setCurrentlyProcessingJobId] =
-    useState<string>()
-  const [currentResponse, setCurrentResponse] = useState('')
-  const [currentResponseError, setCurrentResponseError] = useState('')
+  const internalChatState = useChat(agentSession)
```

Comment:
Use chat will collect get all of the chat state, so all of that logic in this component has been (re)moved and changed.